File: app/src/main/java/coding/yu/helloffmpeg/FfmpegUtils.java
package coding.yu.helloffmpeg;

public class FfmpegUtils {
    private FfmpegUtils() {
    }

    private static FfmpegUtils singleTon = null;

    public static FfmpegUtils instance() {
        if (singleTon == null) {
            synchronized (FfmpegUtils.class) {
                if (singleTon == null)
                    singleTon = new FfmpegUtils();
            }
        }
        return singleTon;
    }

    static {
        System.loadLibrary("native-lib");
    }

    public native String stringFromJNI();

    public native String string2FromJNI();

    public native void initialize();
}

File: app/src/main/java/coding/yu/helloffmpeg/FfmpegUtils.kt
package coding.yu.helloffmpeg

import android.view.Surface

object FfmpegUtils {
    init {
        System.loadLibrary("native-lib")
    }

    external fun stringFromJNI(): String

    external fun release(decoder: Long)

    external fun getDecoder():Long

    external fun decodeStream(data:ByteArray,length:Int,decoder:Long,surface:Surface)


}
